### PR TITLE
Mappings for `net.minecraft.stats`.

### PR DESCRIPTION
**Pull request by Harleyoc1**
Mapped (and updated) all parameters in `net.minecraft.stats`. Also added some Javadoc to stat-related classes (honestly no idea why recipe book classes are in this package).

### CODE REVIEW

File: data/net/minecraft/stats/ServerStatsCounter.mapping [hunk 1]
```diff
@@ -1,7 +1,28 @@
 CLASS net/minecraft/stats/ServerStatsCounter
+	COMMENT Server-side implementation of {@link net.minecraft.stats.StatsCounter}; handles counting, serialising, and
+	COMMENT de-serialising statistics, as well as sending them to connected clients via the
+	COMMENT {@linkplain net.minecraft.network.protocol.game.ClientboundAwardStatsPacket award stats packet}.
```
Comment (alcatrazEscapee) [inline data/net/minecraft/stats/ServerStatsCounter.mapping]: Comment should be one line.

File: data/net/minecraft/stats/Stat.mapping [hunk 2]
```diff
@@ -1,8 +1,30 @@
 CLASS net/minecraft/stats/Stat
+	COMMENT An immutable statistic to be counted for a particular entry in the {@linkplain #type}'s registry. This is
+	COMMENT used as a key in a {@link net.minecraft.stats.StatsCounter} for a corresponding count.
+	COMMENT <p>
+	COMMENT By default, the statistic's {@linkplain #getName() name} is formatted {@code <stat type namespace>.<stat type path>:
+	COMMENT <value namespace>.<value path>}, as created by {@link #buildName(StatType, Object)}.
+	COMMENT
+	COMMENT @param <T> the type of the registry entry for this statistic
+	COMMENT @see net.minecraft.stats.StatType
+	COMMENT @see net.minecraft.stats.Stats
```
Comment (alcatrazEscapee) [inline data/net/minecraft/stats/Stat.mapping]: Incorrect newlines. Don't break in the middle of a sentence. Either use `<p>` on empty lines or don't, don't mix styles.

File: data/net/minecraft/stats/Stat.mapping [hunk 3]
```diff
@@ -1,8 +1,30 @@
...
+		COMMENT The parent statistic type.
+	FIELD value Ljava/lang/Object;
+		COMMENT The registry entry for this statistic.
+	METHOD <init> (Lnet/minecraft/stats/StatType;Ljava/lang/Object;Lnet/minecraft/stats/StatFormatter;)V
+		ARG 1 type
+		ARG 2 value
+		ARG 3 formatter
 	METHOD buildName (Lnet/minecraft/stats/StatType;Ljava/lang/Object;)Ljava/lang/String;
+		COMMENT @return the name for the specified {@code type} and {@code value} in the form {@code <stat type namespace>.<stat
+		COMMENT type path>:<value namespace>.<value path>}
```
Comment (alcatrazEscapee) [inline data/net/minecraft/stats/Stat.mapping]: Comment should be on one line.

File: data/net/minecraft/stats/StatFormatter.mapping [hunk 4]
```diff
@@ -0,0 +1,19 @@
+CLASS net/minecraft/stats/StatFormatter
+	COMMENT A formatter for a {@linkplain Stat statistic}'s corresponding count.
+	COMMENT <p>
+	COMMENT This is a {@linkplain java.lang.FunctionalInterface functional interface} whose functional method is
+	COMMENT {@link #format(int)}.
```
Comment (alcatrazEscapee) [inline data/net/minecraft/stats/StatFormatter.mapping]: Unnecessary comment. This is explaining expected knowledge. (The last two lines specifically)

File: data/net/minecraft/stats/StatType.mapping [hunk 5]
```diff
@@ -1,7 +1,29 @@
...
+	COMMENT items used} or {@link net.minecraft.stats.Stats#BLOCK_MINED the number of blocks mined}. However, there is also a {@link net.minecraft.stats.Stats#CUSTOM
+	COMMENT custom type} which uses entries from the {@linkplain Registry#CUSTOM_STAT custom stat registry}. This is keyed by a
+	COMMENT {@link net.minecraft.resources.ResourceLocation} and can be used to count any statistic that doesn't require an
+	COMMENT associated {@link net.minecraft.core.Registry} entry.
+	COMMENT
+	COMMENT @param <T> the type of the associated registry's entry values
+	COMMENT @see net.minecraft.stats.Stat
+	COMMENT @see net.minecraft.stats.Stats
+	COMMENT @see net.minecraft.core.Registry#STAT_TYPE
+	COMMENT @see net.minecraft.core.Registry#CUSTOM_STAT
```
Comment (alcatrazEscapee) [inline data/net/minecraft/stats/StatType.mapping]: Incorrect usage of newlines in comments.

File: data/net/minecraft/stats/StatsCounter.mapping [hunk 6]
```diff
@@ -1,14 +1,20 @@
 CLASS net/minecraft/stats/StatsCounter
+	COMMENT Manages counting a set of {@link net.minecraft.stats.Stat} objects, stored by a map of statistics to their count.
+	COMMENT <p>
+	COMMENT This base {@code StatsCounter} is only used client-side for keeping track of and reading counts sent from the
+	COMMENT server.
+	COMMENT
+	COMMENT @see net.minecraft.stats.ServerStatsCounter
```
Comment (alcatrazEscapee) [inline data/net/minecraft/stats/StatsCounter.mapping]: Incorrect usage of newlines in comments.

**Review by alcatrazEscapee — CHANGES_REQUESTED** [review]
Mostly just inconsistient use of newlines. In short:

- Don't insert line breaks (new `COMMENT` lines) manually for overlong lines (although do so if you want the comment itself to break a line.
- Use `<p>` for empty lines if you want an explicit newline, or don't (your choice). Don't mix and match styles, especially within the same comment.

**Review by alcatrazEscapee — APPROVED** [review]

**Review by Lanse505 — APPROVED** [review]

**Pull request merged**